engine/src/test/java/io/camunda/zeebe/engine/processing/deployment/model/validation/LinkEventDefinitionTest.java

```diff
@@ -0,0 +1,227 @@
+
+    // then
+    Assertions.assertThat(rejectedDeployment)
+        .hasKey(ExecuteCommandResponseDecoder.keyNullValue())
+        .hasRecordType(RecordType.COMMAND_REJECTION)
+        .hasIntent(DeploymentIntent.CREATE)
+        .hasRejectionType(RejectionType.INVALID_ARGUMENT);
+    assertThat(rejectedDeployment.getRejectionReason())
+        .contains("Element: Gateway_")
+        .contains("ERROR: Event-based gateway must have at least 2 outgoing sequence flows.")
+        .contains(
+            "ERROR: Event-based gateway must not have an outgoing sequence flow to other elements than message/timer intermediate catch events.");
+  }
+
+  @Test
+  public void shouldAcceptDeploymentValidEventLink() throws Exception {
+    // given
+    final Path path =
+        Paths.get(
+            getClass().getResource("/processes/LinkEventTest.testValidEventLink.bpmn").toURI());
+    final byte[] resource = Files.readAllBytes(path);
+
+    // when
+    final Record<DeploymentRecordValue> deployment =
+        ENGINE.deployment().withXmlResource(resource).deploy();
+
+    // then
+    assertThat(deployment.getKey())
+        .describedAs("Should accept deployment valid event link process")
+        .isNotNegative();
+  }
+
+  @Test
+  public void shouldAcceptDeploymentValidEventLinkMultipleSources() throws Exception {
+    // given
+    final Path path =
+        Paths.get(
+            getClass()
+                .getResource("/processes/LinkEventTest.testEventLinkMultipleSources.bpmn")
+                .toURI());
+    final byte[] resource = Files.readAllBytes(path);
+
+    // when
+    final Record<DeploymentRecordValue> deployment =
+        ENGINE.deployment().withXmlResource(resource).deploy();
+
+    // then
+    assertThat(deployment.getKey())
+        .describedAs("Should accept deployment valid event link multiple sources process")
+        .isNotNegative();
+  }
+
+  @Test
+  public void shouldExecuteValidEventLink() throws Exception {
+    // given
+    final Path path =
+        Paths.get(
+            getClass().getResource("/processes/LinkEventTest.testValidEventLink.bpmn").toURI());
+    final byte[] resource = Files.readAllBytes(path);
+
+    ENGINE.deployment().withXmlResource(resource).deploy();
+
+    // when
+    final long processInstanceKey =
+        ENGINE.processInstance().ofBpmnProcessId("Process_05x4y1u").create();
+
+    // then
+    final List<Record<ProcessInstanceRecordValue>> processInstanceEvents =
+        RecordingExporter.processInstanceRecords()
+            .withProcessInstanceKey(processInstanceKey)
+            .limitToProcessInstanceCompleted()
+            .asList();
+
+    assertThat(processInstanceEvents)
+        .extracting(e -> e.getValue().getElementId(), Record::getIntent)
+        .contains(tuple("Process_05x4y1u", ProcessInstanceIntent.COMPLETE_ELEMENT));
+  }
+
+  @Test
+  public void shouldRejectDeployMultipleStartEventsWithSameSignal() {
```

Comment:
:x: We should rename the method to reflect the test case.

The process is rejected because it contains no link catch event for the link name.

```suggestion
  public void shouldRejectDeploymentIfLinkEventNotAppearInPairs() {
```